### PR TITLE
Add automatic provision of the --tag option to npm publish

### LINKED ISSUE #55: Use --prerelease-label to supply the --tag option to npm publish
Setting the `--tag` option to the value of the `--prerelease-label` would ensure that new package versions are automatically published with a reasonable [dist-tag](https://docs.npmjs.com/cli/dist-tag). Publishing a beta version would then *not* override the `@latest` version known to npm.

### PR DESCRIPTION
**Pull request by Avaq**
Closes #55

### CODE REVIEW

**Comment by davidchambers** [comment]
The `"${options[publish-command]}" == 'npm publish'` check doesn't sit well with me. Semantically, I believe we're trying to express the following:

    if pre-release
      default-publish-command = npm publish --tag options[prerelease-label]
    else
      default-publish-command = npm publish

    if user-specified-publish-command
      publish-command = user-specified-publish-command
    else
      publish-command = default-publish-command

The above could instead be expressed like so:

    if user-specified-publish-command
      publish-command = user-specified-publish-command
    else if pre-release
      publish-command = npm publish --tag options[prerelease-label]
    else
      publish-command = npm publish

Rather than set the default value of `publish-command` conditionally, the code in this pull request updates the `publish-command` value if it is `'npm publish'`, regardless of whether the default value was used or whether the user specified `'npm publish'` explicitly.

Although the subtle difference between the two approaches is unlikely to be significant in practice, I would find the code easier to understand if it only manipulated the default `publish-command` value.

**Comment by Avaq** [comment]
I knew you'd say that. I tried going for the suggested approach first, but found that in order to run this logic:

```
if user-specified-publish-command
  publish-command = user-specified-publish-command
else if pre-release
  publish-command = npm publish --tag options[prerelease-label]
else
  publish-command = npm publish
```

Two separate variables are needed (`user-specified-publish-command` and `publish-command`), which needed various changes to the code.

I figured by using the right wording in the documentation, I could get away with the current approach.

**Comment by davidchambers** [comment]
Could we leave `options[publish-command]` unset initially, and set it to the desired default value later if it remains unset?

**Comment by Avaq** [comment]
Sure :)
I thought you'd disapprove because of the mixed type, in that case.

**Comment by Avaq** [comment]
I just released a new beta version of Fluture using this version of xyz. I'm happy to report that it works as intended. :)

**Comment by Avaq** [comment]
Shall we merge and release this?

**Comment by davidchambers** [comment]
> Shall we merge and release this?

Yes. Thank you for reminding me. `v3.1.0` or `v4.0.0`?

**Comment by Avaq** [comment]
> `v3.1.0` or `v4.0.0`?

Technically I think this is a breaking change, because the behaviour changed. I would be surprised if anyone relies on beta packages being published with the `latest` tag, but I still think it's good to do a major bump, if only to give people a reason to look at what's changed.

File: xyz [hunk 1]
```diff
@@ -34,13 +35,17 @@ Options:
    --prerelease-label <label>
         Specify the label to be used in the version number when publishing
         a pre-release version (e.g. 'beta' is the label in '2.0.0-beta.0').
-        'rc' is assumed if this option is omitted.
+        'rc' is assumed if this option is omitted. If the release is a
+        prerelease, as indicated by --increment, the --prerelease-label will
```
Comment (davidchambers) [inline xyz]: We're using *pre-release* (hyphenated) two lines above. To be consistent, let's hyphenate the word in contexts in which it does not have special meaning.

**Comment by Avaq** [comment]
@davidchambers Would you like me to merge, and perhaps release, this?

**Comment by davidchambers** [comment]
Absolutely. In the future please let me know when you `git push --force` a change as I rely on GitHub notifications to know when action is required from me. I'm sorry to have left you hanging.

**Comment by Avaq** [comment]
Hi @davidchambers. Any plans to release this to npm? I've been using direct install from GitHub all this time.

**Comment by davidchambers** [comment]
<https://github.com/davidchambers/xyz/releases/tag/v4.0.0>

Sorry for the delay.

**Pull request merged**